aten/src/ATen/native/Activation.cpp

```diff
@@ -43,6 +43,205 @@ Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Genera
+      });
+    });
+  }
+  else { // case2: multiple weights, one for each channel
+    int64_t input_ndim = input.dim();
+    AT_CHECK(input_ndim > 0, "Not allow zero-dim input tensor.");
+
+    int64_t channel_size = 1; // channel_size default to 1
+    int64_t input_dim0_size = 1, input_stride0 = 1, input_stride1 = 1;
+
+    if (input_ndim > 1) {
+      channel_size = input.size(1); // channel is the 2nd dim of input
+      input_dim0_size = input.size(0);
+      input_stride0 = strides[0];
+      input_stride1 = strides[1];
+    }
+    AT_CHECK(channel_size == weight_num,
+      "Mismatch of parameter numbers and input channel size. Found parameter numbers = %d, and channel size = %d.",
+      weight_num, channel_size);
+
+    AT_DISPATCH_FLOATING_TYPES(input.type(), "prelu_cpu", [&] {
+      prelu_cpu_kernel<scalar_t>(
+        result,
+        input,
+        weight,
+        input_dim0_size,
+        channel_size,
+        input_stride0,
+        input_stride1);
+    });
+  }
+  return result;
+}
+
+// -----------------------------------
+// prelu backward
+// -----------------------------------
+template <typename scalar_t>
+void prelu_cpu_backward_kernel(
+  const Tensor& input,
+  const Tensor& weight,
+  const Tensor& grad_out,
+  Tensor& input_grad,
+  Tensor& weight_grad_collector,
+  int64_t input_dim0_size,
+  int64_t channel_size,
+  int64_t input_stride0,
+  int64_t input_stride1) {
+
+  int64_t i, j, k;
+  auto input_data = input.data<scalar_t>();
+  auto weight_data = weight.data<scalar_t>();
+  auto grad_out_data = grad_out.data<scalar_t>();
+  auto input_grad_data = input_grad.data<scalar_t>();
+  auto weight_grad_collector_data = weight_grad_collector.data<scalar_t>();
+
+  #pragma omp parallel for private(i, j, k) if (input.numel() > 10000)
+  for (i = 0; i < input_dim0_size; i++) {
+    for (j = 0; j < channel_size; j++) {
+      for (k = 0; k < input_stride1; k++) {
+        int64_t pos = i * input_stride0 + j * input_stride1 + k;
+        input_grad_data[pos] = (input_data[pos] > 0) ? grad_out_data[pos] : weight_data[j] * grad_out_data[pos];
+        weight_grad_collector_data[pos] = (input_data[pos] > 0) ? scalar_t(0) : input_data[pos] * grad_out_data[pos];
+      }
+    }
+  }
+}
+
+std::tuple<Tensor, Tensor> prelu_backward_cpu(const Tensor& grad_out_, const Tensor& self, const Tensor& weight_) {
+  auto input = self.contiguous();
+  auto grad_out = grad_out_.contiguous();
+  auto weight = weight_.contiguous();
+
+  AT_CHECK(input.is_contiguous());
+  AT_CHECK(grad_out.is_contiguous());
+  AT_CHECK(weight.is_contiguous());
+
+  int64_t weight_num = weight.numel();
+  auto strides = input.strides();
+  auto dims = input.dim();
+
+  Tensor input_grad = at::empty_like(input);
+  Tensor weight_grad = at::empty_like(weight);
+  Tensor weight_grad_collector = at::empty_like(input);
+
+  // case1: shared parameter for all channels
+  if (weight_num == 1) {
```

Comment:
CPU backward should also be similar to previous implementation:
https://github.com/pytorch/pytorch/blob/3da8d71d7de6f2eb3c9bbe395e38dc2fdf2784b0/aten/src/THNN/generic/PReLU.c#L71-L81

and 
https://github.com/pytorch/pytorch/blob/3da8d71d7de6f2eb3c9bbe395e38dc2fdf2784b0/aten/src/THNN/generic/PReLU.c#L146-L156